samples/navigation-fragments/src/main/java/com/instacart/formula/navigation/CounterFragmentFeatureFactory.kt

```diff
@@ -0,0 +1,34 @@
+package com.instacart.formula.navigation
+
+import com.instacart.formula.android.Feature
+import com.instacart.formula.android.FeatureFactory
+import com.instacart.formula.runAsStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+
+class CounterFragmentFeatureFactory : FeatureFactory<CounterFragmentFeatureFactory.Dependencies, CounterFragmentKey>() {
+
+    interface Dependencies {
+        fun navigationStack(): List<Int>
+        fun counterIncrements(): SharedFlow<Int>
+        fun onNavigateToNext(): () -> Unit
+        fun onNavigateBack(): () -> Unit
+        fun onIncrementCounter(): (Int) -> Unit
+    }
+
+    override fun Params.initialize(): Feature {
+        return Feature(CounterFragmentViewFactory()) {
+            val formula = CounterFragmentFormula()
+            formula.runAsStateFlow(
+                scope = it,
+                input = CounterFragmentFormula.Input(
+                    fragmentId = key.fragmentId,
+                    navigationStack = dependencies.navigationStack(),
```

Comment:
I'm not a fan of pulling `navigationStack` like this. It's hard to understand if this will work with process restoration. Instead, I would keep a counter index on each fragment key and increment that when creating the new key. Then, each counter can subscribe to navigation state events and only take data up to the index.